compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -246,26 +248,43 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
                 } else {
                     None
                 };
-                (String::new(), "this scope".to_string(), suggestion)
+                (String::new(), "this scope".to_string(), None, suggestion)
             } else if path.len() == 2 && path[0].ident.name == kw::PathRoot {
                 if self.r.tcx.sess.edition() > Edition::Edition2015 {
                     // In edition 2018 onwards, the `::foo` syntax may only pull from the extern prelude
                     // which overrides all other expectations of item type
                     expected = "crate";
-                    (String::new(), "the list of imported crates".to_string(), None)
+                    (String::new(), "the list of imported crates".to_string(), None, None)
                 } else {
-                    (String::new(), "the crate root".to_string(), None)
+                    (
+                        String::new(),
+                        "the crate root".to_string(),
+                        Some(DefId::local(rustc_span::def_id::CRATE_DEF_INDEX)),
```

Comment:
```suggestion
                        Some(CRATE_DEF_ID),
```
And same below.

Reply:
This isn't entirely possible as it's a `DefId` in another branch and then used as that later, but I changed it to `CRATE_DEF_ID.to_def_id()`.